subsys/net/ip/net_shell.c

```diff
@@ -928,6 +940,89 @@ int net_shell_cmd_dns(int argc, char *argv[])
 	return 0;
 }
 
+#if defined(CONFIG_NET_DEBUG_HTTP_CONN) && defined(CONFIG_HTTP_SERVER)
+#define MAX_HTTP_OUTPUT_LEN 64
+
+static char *http_str_output(char *output, int outlen, const char *str, int len)
+{
+	if (len > outlen) {
+		len = outlen;
+	}
+
+	if (len == 0) {
+		memset(output, 0, outlen);
+	} else {
+		memcpy(output, str, len);
+		output[len] = '\0';
+	}
+
+	return output;
+}
+
+static void http_server_cb(struct http_server_ctx *entry,
+			   void *user_data)
+{
+	int *count = user_data;
+	static char output[MAX_HTTP_OUTPUT_LEN];
+
+	/* +7 for []:port */
+	char addr_local[ADDR_LEN + 7];
+	char addr_remote[ADDR_LEN + 7] = "";
+
+	get_addresses(entry->req.net_ctx, addr_local, sizeof(addr_local),
+		      addr_remote, sizeof(addr_remote));
+
+	if (*count == 0) {
+		printk("      HTTP ctx    Local           \t"
+		       "Remote          \tURL\n");
+	}
+
+	(*count)++;
+
+	printk("%s[%2d]%c %p  %16s\t%16s\t%s\n",
+	       !entry->enabled ? "-" : "", *count,
```

Comment:
BTW, I did not print the Unicode lock symbol as that might not be printed properly always.